ReSwift/CoreTypes/Subscription.swift

```diff
@@ -8,7 +8,138 @@
+    weak var subscriber: AnyStoreSubscriber?
+
+    init<T>(
+        originalSubscription: Subscription<State>,
+        transformedSubscription: Subscription<T>?,
+        subscriber: AnyStoreSubscriber
+    ) {
+        self.originalSubscription = originalSubscription
+        self.subscriber = subscriber
+
+        // If we received a transformed subscription, we subscribe to that subscription
+        // and forward all new values to the subscriber.
+        if let transformedSubscription = transformedSubscription {
+            transformedSubscription.observe { oldState, newState in
+                self.subscriber?._newState(state: newState as Any)
+            }
+        // If we haven't received a transformed subscription, we forward all values
+        // from the original subscription.
+        } else {
+            originalSubscription.observe { oldState, newState in
+                self.subscriber?._newState(state: newState as Any)
+            }
+        }
+    }
+
+    func newValues(oldState: State, newState: State) {
+        // We pass all new values through the original subscription, which accepts
+        // values of type `<State>`. If present, transformed subscriptions will
+        // receive this update and transform it before passing it on to the subscriber.
+        self.originalSubscription.newValues(oldState: oldState, newState: newState)
+    }
+}
+
+/// Represents a subscription of a subscriber to the store. The subscription determines which new
+/// values from the store are forwarded to the subscriber, and how they are transformed.
+/// The subscription acts as a very-light weight signal/observable that you might know from
+/// reactive programming libraries.
+public class Subscription<State> {
+
+    private func _select<Substate>(
+        _ selector: @escaping (State) -> Substate
+        ) -> Subscription<Substate>
+    {
+        return Subscription<Substate> { sink in
+            self.observe { oldState, newState in
+                sink(oldState.map(selector) ?? nil, newState.map(selector) ?? nil)
+            }
+        }
+    }
+
+    // MARK: Public Interface
+
+    /// Provides a subscription that selects a substate of the state of the original subscription.
+    /// - parameter selector: A closure that maps a state to a selected substate
+    public func select<Substate>(
+        _ selector: @escaping (State) -> Substate
+        ) -> Subscription<Substate>
+    {
+        return self._select(selector)
+    }
+
+    /// Provides a subscription that selects a substate of the state of the original subscription.
+    /// If the selected substate is `Equatable` repeated state updates will be skipped.
+    /// - parameter selector: A closure that maps a state to a selected substate
+    public func select<Substate: Equatable>(
+        _ selector: @escaping (State) -> Substate
+        ) -> Subscription<Substate>
+    {
+        return self._select(selector).skipRepeats()
+    }
```

Comment:
@mjarvis @DivineDominion This is where the default implementation of `skipRepeats` for `Equatable` substates lives.